Expose the functions of ruby's built-in Shellwords module

Joris-van-der-Wel opened the pull request:
This change adds three new functions based on [Shellwords](http://ruby-doc.org/stdlib-2.3.0/libdoc/shellwords/rdoc/Shellwords.html):

* `bourne_shell_escape(string)` (`Shellwords.shellescape`)
* `bourne_shell_join(array)` (`Shellwords.shelljoin`)
* `bourne_shell_split(string)` (`Shellwords.shellsplit`)

I seem to be repeating these escape and join functions in a lot of my modules. I've also hit shell escaping issues while using a few popular modules from the forge. I've included the split function for the sake of completeness.

hunner commented on README.markdown:
```diff
@@ -206,6 +206,40 @@ bool2str(false, 't', 'f')         => 'f'
 
 Requires a single boolean as input. *Type*: rvalue.
 
+#### `bourne_shell_escape`
```
Could you actually just name these `shell_*` instead of `bourne_shell_*` ? It's kind of assumed that unless explicitly mentioned like "ksh", "csh", "bash" that "shell" means "bourne".

hunner commented on README.markdown:
```diff
@@ -206,6 +206,40 @@ bool2str(false, 't', 'f')         => 'f'
 
+#### `bourne_shell_escape`
+
+Escapes a string so that it can be safely used in a Bourne shell command line. Note that the resulting string should be used unquoted and is not intended for use in double quotes nor in single quotes. This function behaves the same as ruby's `Shellwords.shellescape()` function.
```
Could the ruby method description link to the ruby docs?

igalic commented:
tests are failing:
```
Failures:

  1) bourne_shell_escape stringification should run bourne_shell_escape(10) and return "10"
     Failure/Error: it { is_expected.to run.with_params(10).and_return('10') }
       expected bourne_shell_escape(10) to have returned "10" instead of raising NoMethodError(undefined method `empty?' for 10:Fixnum)
     # ./spec/functions/bourne_shell_escape_spec.rb:12:in `block (3 levels) in <top (required)>'

  2) bourne_shell_escape stringification should run bourne_shell_escape(false) and return "false"
     Failure/Error: it { is_expected.to run.with_params(false).and_return('false') }
       expected bourne_shell_escape(false) to have returned "false" instead of raising NoMethodError(undefined method `empty?' for false:FalseClass)
     # ./spec/functions/bourne_shell_escape_spec.rb:13:in `block (3 levels) in <top (required)>'
```

Joris-van-der-Wel commented:
Updated:
* removed the bourne_ prefix from the functions
* ruby 1.9 fix (the failing tests mentioned earlier)
* links to ruby doc
* rebased on master

By the way, there is a test failing unrelated to my changes (it is also failing in other PR's):

```
1) get_module_path should run get_module_path("one") and raise an Puppet::Environments::EnvironmentNotFound with the message matching /Could not find a directory environment/

     Failure/Error: it { is_expected.to run.with_params('one').and_raise_error(Puppet::Environments::EnvironmentNotFound, /Could not find a directory environment/) }

       expected get_module_path("one") to have raised Puppet::Environments::EnvironmentNotFound matching /Could not find a directory environment/ instead of Puppet::ParseError(Could not find module one in environment rp_env)

     # ./spec/functions/get_module_path_spec.rb:9:in `block (2 levels) in <top (required)>'
```

tphoney commented:
Thanks for the PR @Joris-van-der-Wel
